assets/app/app.js

```diff
@@ -12,20 +12,39 @@ var encodeB64 = require('./helpers/encoding').encodeB64;
 
 var Router = Backbone.Router.extend({
   initialize: function () {
-    var self = this;
 
-    this.sites = new SiteCollection();
     this.user = new UserModel();
 
-    this.navbarView = new NavbarView({ model: this.user });
-    this.mainView = new MainContainerView({ user: this.user, collection: this.sites });
 
-    this.listenToOnce(this.user, 'change', function () {
+    if (window.localStorage.getItem('token')) {
+      $('#home').hide();
+      $('#main-loader').show();
+    }
+
+    this.listenToOnce(this.user, 'sync', function (user) {
       var token = this.user.attributes.passports[0].tokens.accessToken;
       window.localStorage.setItem('token', encodeB64(token));
-      self.navbarView.render();
-      Backbone.history.loadUrl();
-    });
+      this.sites = new SiteCollection();
+      this.sites.fetch({
+        data: $.param({ limit: 50 }),
+        success: function (sites) {
+          this.navbarView = new NavbarView({ model: this.user });
+          this.mainView = new MainContainerView({
+            user: this.user,
+            collection: this.sites
+          });
+          this.navbarView.render();
+          Backbone.history.start();
+        }.bind(this)
+      });
+    }.bind(this));
```

Comment:
I think the `var self` trick works better here, otherwise adding an additional nested function requires `bind` which seems error prone to me.